packages/codec/lib/abi-data/allocate/index.ts

```diff
@@ -650,22 +721,33 @@ function getCalldataAllocationsForContract(
 }
 
 //note: returns undefined if undefined is passed in
+//for first argument
 function defaultConstructorAllocation(
-  constructorContext: Contexts.DecoderContext
-): CalldataAllocation | undefined {
+  constructorContext: Contexts.DecoderContext,
+  deployedContext: Contexts.DecoderContext
+): CalldataAndReturndataAllocation | undefined {
   if (!constructorContext) {
     return undefined;
   }
   let rawLength = constructorContext.binary.length;
   let offset = (rawLength - 2) / 2; //number of bytes in 0x-prefixed bytestring
-  return {
+  let input = {
     offset,
     abi: AbiDataUtils.DEFAULT_CONSTRUCTOR_ABI,
     arguments: [] as CalldataArgumentAllocation[],
-    allocationMode: "full"
+    allocationMode: "full" as const
   };
+  let output = CONSTRUCTOR_OUTPUT_ALLOCATION;
+  return { input, output };
 }
 
+const CONSTRUCTOR_OUTPUT_ALLOCATION: ReturndataAllocation = {
```

Comment:
Seems that we might as well name and export this like the various other functions here... so what if it's a function that takes zero arguments?

Something like:
```suggestion
export const constructorOutputAllocation: ReturndataAllocation = {
```

Reply:
Wait, do you want it as an object or literally as a function that takes zero arguments?  Yes I'm aware that in mathematics or Haskell they are the same but this is Typescript so the distinction matters...

Reply:
Also, to what extent should this apply to other constants?

Reply:
> Also, to what extent should this apply to other constants?

I think it should apply to all constants, really. I know I set the precedent here, but while doing the codec reorg, the UPPER_SNAKE_EVERYWHERE really seemed out of place, stylistically.

Reply:
OK, did this, but note I only changed this for new constants; I left existing ones alone.  Let me know if that should be changed more generally and I can make a separate PR (although there's the question of to what extent that would be breaking).

Reply:
For anything in the codec, I think it's fine, since all minor releases will also be breaking according to semver. I think we can probably just do this as we spot individual ones.